From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Wolfgang Bumiller <w.bumiller@proxmox.com>
Date: Mon, 6 Apr 2020 12:16:53 +0200
Subject: [PATCH] PVE: monitor: disable oob capability

A bisect revealed that commit 8258292e18c3
("monitor: Remove "x-oob", offer capability "oob" unconditionally")
causes unexpected hangs when restoring live snapshots from some
types of block devices (particularly RBD).
We need to figure out what's happnening there. For now, since we
had this disabled before and probably don't need it now either,
disable oob, so we can get a functioning qemu out...

Signed-off-by: Wolfgang Bumiller <w.bumiller@proxmox.com>
Signed-off-by: Thomas Lamprecht <t.lamprecht@proxmox.com>
[FE: rebase for 11.1.0]
Signed-off-by: Fiona Ebner <f.ebner@proxmox.com>
---
 monitor/qmp.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/monitor/qmp.c b/monitor/qmp.c
index 681533715f..31f866a4f6 100644
--- a/monitor/qmp.c
+++ b/monitor/qmp.c
@@ -174,8 +174,7 @@ static void monitor_qmp_emit_event(Monitor *mon, QAPIEvent event, QDict *qdict)
 
 static bool monitor_qmp_requires_iothread(const Monitor *mon)
 {
-    return qemu_chr_has_feature(mon->chr.chr,
-                                QEMU_CHAR_FEATURE_GCONTEXT);
+    return false;
 }
 
 static bool qmp_oob_enabled(MonitorQMP *mon)
